ios/RNInstabug/InstabugNetworkLoggerBridge.m

```diff
@@ -44,6 +73,145 @@ -(void)stopObserving {
     resolve(@(IBGNetworkLogger.isNativeNetworkInterceptionFeatureEnabled));
 }
 
+RCT_EXPORT_METHOD(registerNetworkLogsListener: (NetworkListenerType) listenerType) {
+    switch (listenerType) {
+         case NetworkListenerTypeFiltering:
+             NSLog(@"Andrew: Network logs filtering enabled");
+             [self setupRequestFilteringHandler];
+             break;
+
+         case NetworkListenerTypeObfuscation:
+             NSLog(@"Andrew: Network logs obfuscation enabled");
+             [self setupRequestObfuscationHandler];
+             break;
+
+         case NetworkListenerTypeBoth:
+             NSLog(@"Andrew: Both filtering and obfuscation enabled");
+             [self setupRequestFilteringAndObfuscationHandler];
+             break;
+
+         default:
+             NSLog(@"Andrew: Unknown NetworkListenerType");
+             break;
+     }
+}
+
+
+RCT_EXPORT_METHOD(updateNetworkLogSnapshot:(NSString * _Nonnull)jsonString) {
+    // Properly initialize the NSMutableURLRequest
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] init];
+
+    // Convert jsonString to NSData
+    NSData *data = [jsonString dataUsingEncoding:NSUTF8StringEncoding];
+
+    // Parse the JSON into a dictionary
+    NSError *error = nil;
+    NSDictionary *dict = [NSJSONSerialization JSONObjectWithData:data options:0 error:&error];
+
+    // Check for JSON parsing errors
+    if (error) {
+        NSLog(@"Failed to parse JSON: %@", error);
+        return;
+    }
+
+    // Set the URL, HTTP body, and headers
+    request.URL = [NSURL URLWithString:dict[@"url"]];
+    request.HTTPBody = [dict[@"requestBody"] dataUsingEncoding:NSUTF8StringEncoding];
+
+    // Ensure requestHeader is a dictionary
+    if ([dict[@"requestHeader"] isKindOfClass:[NSDictionary class]]) {
+        request.allHTTPHeaderFields = dict[@"requestHeader"];
+    } else {
+        NSLog(@"Invalid requestHeader format");
+//        return;
+    }
+
+    // Ensure self.completion is not nil before calling it
+    NSString *callbackID = dict[@"id"];
+    if ([callbackID isKindOfClass:[NSString class]] && self.requestObfuscationCompletionDictionary[callbackID] != nil) {
+        ((IBGURLRequestAsyncObfuscationCompletedHandler)self.requestObfuscationCompletionDictionary[callbackID])(request);
+    } else {
+        NSLog(@"Not Available Completion");
+    }
+
+}
+
+RCT_EXPORT_METHOD(setNetworkLoggingRequestFilterPredicateIOS: (NSString * _Nonnull) callbackID : (BOOL)value ){
+
+    if ([callbackID isKindOfClass:[NSString class]] && self.requestFilteringCompletionDictionary[callbackID] != nil) {
```

Comment:
I assume `callbackID` will always be a non-null `NSString *` given the method signature, why do we need the check for its class? 🤔
```suggestion
    if (self.requestFilteringCompletionDictionary[callbackID] != nil) {
```

Reply:
This code snippet was created by **Eyad** from the iOS team, and this is how it worked for them.

Reply:
I confirmed with Eyad and he agreed that it's not needed. I think we can remove it.